lib/common/compiler.h

```diff
@@ -92,7 +92,7 @@
 #if defined(_MSC_VER) && (defined(_M_X64) || defined(_M_I86))  /* _mm_prefetch() is not defined outside of x86/x64 */
 #  include <mmintrin.h>   /* https://msdn.microsoft.com/fr-fr/library/84szxsww(v=vs.90).aspx */
 #  define PREFETCH(ptr)   _mm_prefetch((const char*)ptr, _MM_HINT_T0)
-#elif defined(__GNUC__)
+#elif defined(__GNUC__) && ( (__GNUC__ >= 4) || ( (__GNUC__ == 3) && (__GNUC_MINOR__ >= 1) ) )
```

Comment:
Does this still work for clang?

Reply:
This change only impacts compilers which define `__GNUC__`, and since I guess clang does not define it, I think we are safe here.
That said, I might be wrong about clang not defining `__GNUC__`. Maybe it does, although this would be counter-intuitive.

Reply:
> This change only impacts compilers which define __GNUC__

To be more precise, I should say that this change is neutral to all compilers which do not define `__GNUC__`.

Reply:
It should work for clang https://godbolt.org/g/yaJ8Q5, it sets `__GNUC__ == 4` since clang-3.0 at least, which is good enough.

Reply:
@fbrosson clang pretends to be gcc, and it supports all or nearly all gcc extensions.